docs/source/4git_users.rst

```diff
@@ -0,0 +1,108 @@
+DataLad For git/git-annex users
+*******************************
+
+Overview
+========
+
+DataLad could be considered a "thin layer" on top of ``git`` and ``git-annex``
+with the primary goal to provide a unified and simplified interface to
+manipulate git/git-annex repositories.  DataLad does neither stores any additional
+"state" information nor provides additional to `.git/objects` and
+`.git/annex/objects` "content store". So, at large, you can work with DataLad
+datasets as with any other git/git-annex repository, or you can use DataLad to work
+with regular git/git-annex repositories.  DataLad extensively relies on standard
+``git submodule`` to manage collections/hierarchies of the repositories
+(called :term:`superdataset` in DataLad terms). So you could even use
+`git submodule` commands directly.
+
+Having said all that DataLad adds and uses some additional information to git/git-annex
+repositories:
+
+- DataLad uses plain `git config` for managing configuration/settings, but in addition
+  to regular configuration file locations (such as `.git/config`) it also considers
+  `.datalad/config` which is also kept under git.
+- The main difference of DataLad :term:`dataset` from a regular git/git-annex
+  repository is that each dataset having a unique identifier (UUID) assigned and committed
+  (by `datalad create`) within `.datalad/config` (``datalad.dataset.id``).
+- DataLad's meta-data "facilities" store extracted and aggregated metadata
+  under `.datalad/metadata`.
+
```

Comment:
We do have issues when working with non-dataset repositories. Even with old datalad datasets that lack the default .gitattributes config. Stuff ends up being annexed, sometimes unintentionally and annoyingly.

Reply:
that is worth a note indeed, something like

DataLad `create` adds "safe-guards" in the configuration to streamline working with git-annex, e.g. to ensure that git configuration files do not get added to the annex etc.